Lib/test/test_interpreters.py

```diff
@@ -0,0 +1,88 @@
+import interpreters
+import os
+from textwrap import dedent
+import unittest
+
+def _captured_script(script):
+    r, w = os.pipe()
+    indented = script.replace('\n', '\n                ')
+    wrapped = dedent(f"""
+        import contextlib
+        with open({w}, 'w') as spipe:
+            with contextlib.redirect_stdout(spipe):
+                {indented}
+        """)
+    return wrapped, open(r)
+
+def clean_up_interpreters():
+    for id in interpreters.list_all():
+        if id == 0:  # main
+            continue
+        try:
+            interpreters.destroy(id)
+        except RuntimeError:
+            pass  # already destroyed
+
+
+class TestBase(unittest.TestCase):
+
+    def tearDown(self):
+        clean_up_interpreters()
+
+
+class ListAllTests(TestBase):
+
+    def test_initial(self):
+        main = interpreters.get_main()
+        ids = interpreters.list_all()
+        self.assertEqual(ids, [main])
+
+
+class GetCurrentTests(TestBase):
+
+    def test_get_current(self):
+        main = interpreters.get_main()
+        cur = interpreters.get_current()
+        self.assertEqual(cur, main)
+
+
+class GetMainTests(TestBase):
+
+    def test_get_main(self):
+        [expected] = interpreters.list_all()
+        main = interpreters.get_main()
+        self.assertEqual(main, expected)
+
+
+class CreateTests(TestBase):
+
+    def test_create(self):
+        id = interpreters.create()
+        self.assertIn(id, interpreters.list_all())
+
```

Comment:
Make sure to add a function test that creates more than one.  You could do that in the existing test.